purescript.cabal

```diff
@@ -144,6 +145,7 @@ common defaults
     Glob >=0.10.1 && <0.11,
     haskeline >=0.8.2 && <0.9,
     language-javascript ==0.7.0.0,
+    lens >=4.19.2 && <4.20,
```

Comment:
This is tangential to the PR, but I thought I'd still bring it up. We already depend on `microlens`. By adding yet another `lens` library as a dependency, it makes the binary larger. I'm assuming that it's being added here because `microlens` doesn't suffice. And if that's the case, then could we refactor the codebase to remove `microlens` so that we only depend on one `lens` library?

Reply:
`monoidal-containers` (which I added in #4283 before realizing this fact) already depends on `lens`; if `lens` and its dependencies (most of which we were already pulling in even before `monoidal-containers`) result in too much binary bloat, I'll need to inline the parts of `monoidal-containers` we're using. I don't actually know how much of a difference in binary size we're talking about here, with all the optimizations on; do you?

We can absolutely replace the various `microlens` usages with `lens`—I've done so in a local branch—but I figured I'd submit that in a separate PR if we merge this deciding we want to keep `lens`.

Reply:
My comment comes mainly from reading a blogpost a couple of years ago pointing out an issue like this: a project depends on a small lens library but transitively depends on all the other lens libraries such that the author just wondered why one didn't just depend on `lens` and call it a day. (Or something like that).

Could you clarify for me what the difference between `microlens` and `lens` is (if any) besides `microlens` just being smaller? For example, do they use the same underlying encoding for optics?

Reply:
My understanding, based on little more than a skimming of `microlens`'s README and some unattributable background impressions, is that the implementation of `microlens` is more or less directly lifted from `lens`; the main if not only difference is how much API and instances are provided and how many dependencies are taken.